### PR TITLE
Deploy storage refactoring

### LINKED ISSUE #3537: Implement new findDeploy API method
## Overview

Current **limitation** of `/api/deploy/{deployId}` API
1. You can not know whether the deploy is received by the validator if there are some network hiccups happen in sending deploy.
2. You can not know why your deploy is not packed in a block even the sending process is successfully.(valid after block number is not valid)

In short, users find no way to know the real status of a deploy that is not in a block. And **the users want to know that.**

### Impact

No impact.

### Design

Provide a `Status` field in the return of the api.
```scala
final case class FindDeployResponse(
  status: InTransactionPools || ValidAfterNumberOutAge || WaitingForFinalization ||Finalized || NotFinalized(Orphaned)
  data: {
    "deployInfo": deployDataTye,
    "blockInfo": blockData,  // exist only the deploy is packed
  }
)
```

## Solution

- [x] The first part is to expose deploy status method on the API.
  - #3682
  - #3695

- [ ] What's left is to add _Expired_, _Finalized_  and _Rejected_ statuses in which deploy can be.

### PR DESCRIPTION
**Pull request by nzpr**
## Overview
This refactoring is a preparation to follow up PR to resolve <https://github.com/rchain/rchain/issues/3537>.
1. Fix blockStore package name, rename create `method` to `apply`
2. Remove IndexBlockDagStorage which is used only in tests.
3. Remove DeployStorage, make it part of BlockDagStorage.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala [hunk 1]
```diff
@@ -122,7 +122,7 @@ class BlocksResponseAPITest
         genesis        <- createDagWith8Blocks(blockStore, blockDagStorage)
         blockApi       <- createBlockApi[Task](genesis.shardId, maxBlockLimit)
         blocksResponse <- blockApi.getBlocks(2)
-      } yield blocksResponse.right.get.length should be(2) // TODO: Switch to 3 when we implement block height correctly
+      } yield blocksResponse.right.get.length should be(3)
```
Comment (nzpr) [inline casper/src/test/scala/coop/rchain/casper/api/BlocksResponseAPITest.scala:120]: This DAG is created https://github.com/rchain/rchain/blob/dev/docs/casper/images/no_finalizable_block_mistake_with_no_disagreement_check.png
so length should be 3. Not sure how it were passing with 2, looks like there were a bug in IndexedBlockDagStorage.

File: block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala [hunk 2]
```diff
@@ -19,6 +20,8 @@ trait BlockDagStorage[F[_]] {
   ): F[Unit]
   def lookup(blockHash: BlockHash): F[Option[BlockMetadata]]
   def lookupByDeployId(blockHash: DeployId): F[Option[BlockHash]]
+  def addDeploy(d: Signed[DeployData]): F[Unit]
+  def pooled: F[Map[DeployId, Signed[DeployData]]]
```
Comment (tgrospic) [inline block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala]: Please put _deploy_ in the name of this function.

File: casper/src/main/scala/coop/rchain/casper/storage/RNodeKeyValueStoreManager.scala [hunk 3]
```diff
@@ -20,9 +20,9 @@ object RNodeKeyValueStoreManager {
   private val evalHistoryEnvConfig = LmdbEnvConfig(name = "eval/history", maxEnvSize = 1 * tb)
   private val evalColdEnvConfig    = LmdbEnvConfig(name = "eval/cold", maxEnvSize = 1 * tb)
   // Blocks
-  private val blockStorageEnvConfig  = LmdbEnvConfig(name = "blockstorage", maxEnvSize = 1 * tb)
-  private val dagStorageEnvConfig    = LmdbEnvConfig(name = "dagstorage", maxEnvSize = 100 * gb)
-  private val deployStorageEnvConfig = LmdbEnvConfig(name = "deploystorage", maxEnvSize = 1 * gb)
+  private val blockStorageEnvConfig = LmdbEnvConfig(name = "blockstorage", maxEnvSize = 1 * tb)
+  private val dagStorageEnvConfig   = LmdbEnvConfig(name = "dagstorage", maxEnvSize = 100 * gb)
+  private val deployPoolEnvConfig   = LmdbEnvConfig(name = "deploypool", maxEnvSize = 1 * gb)
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/storage/RNodeKeyValueStoreManager.scala]: Why not call deploy DB file in the same way as other files?

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/storage/RNodeKeyValueStoreManager.scala]: IMO `deploystorage` tells that this is a general storage for all deploys, not a pool of deploys submitted to validator.  I would expect that `deploystorage` is a storage of deploys coming with blocks.

File: block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala (hunk 2 above)
Comment (nzpr) [reply block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala]: https://github.com/rchain/rchain/pull/3673/commits/3e8d3da8f2ea9f7c5fb8a030ebbc4cc203006dfa

File: casper/src/main/scala/coop/rchain/casper/storage/RNodeKeyValueStoreManager.scala (hunk 3 above)
Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/storage/RNodeKeyValueStoreManager.scala]: Renamed to `deploypoolstorage` in [Rename pooled](https://github.com/rchain/rchain/pull/3673/commits/b4b4ab17ede42f7d141717aa32d0a9b9cd246d1a)

**Review by tgrospic — APPROVED** [review]
LGTM

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**